kernel/kernel-api/src/main/java/io/delta/kernel/internal/checksum/CRCInfo.java

```diff
@@ -82,6 +86,14 @@ public static Optional<CRCInfo> fromColumnarBatch(
         txnIdColumnVector.isNullAt(rowId)
             ? Optional.empty()
             : Optional.of(txnIdColumnVector.getString(rowId));
+    ColumnVector domainMetadataVector = batch.getColumnVector(getSchemaIndex(DOMAIN_METADATA));
```

Comment:
If load is gonna be in follow up PR shouldn't we not read it in `fromColumnarBatch`?

Reply:
Nvm, I see that we are updating the tests for CheckSumWriter. But doesn't this PR contain all the changes to read DM from CRC? We just don't write it with Kernel yet so it is always empty?

Reply:
I actually also think we don't need this in the PR? But if we do can you update description that we are also adding read support?

Reply:
Done, Replace this with a TODO.